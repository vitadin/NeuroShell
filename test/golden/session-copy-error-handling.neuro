# Test session copy error handling
\session-new[system="You are a test assistant"] valid_session

# Test 1: No source parameters
\try \session-copy[target_session_name=should_fail]
\echo Error 1: ${@error}

# Test 2: Both source parameters (should fail)
\try \session-copy[source_session_id=abc123, source_session_name=valid_session, target_session_name=should_fail]
\echo Error 2: ${@error}

# Test 3: Nonexistent source session
\try \session-copy[source_session_name=nonexistent_session, target_session_name=should_fail]
\echo Error 3: ${@error}

# Test 4: Duplicate target name
\session-new existing_target
\try \session-copy[source_session_name=valid_session, target_session_name=existing_target]
\echo Error 4: ${@error}