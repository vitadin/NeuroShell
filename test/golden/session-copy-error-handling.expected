# Test session copy error handling
Created session 'valid_session' (ID: 00000001)
# Test 1: No source parameters
Error 1: must specify either source_session_id or source_session_name
# Test 2: Both source parameters (should fail)
Error 2: cannot specify both source_session_id and source_session_name - use exactly one
# Test 3: Nonexistent source session
Error 3: session copy failed: source session lookup failed: no session found for 'nonexistent_session' (tried exact name, exact ID, prefix match)
# Test 4: Duplicate target name
Created session 'existing_target' (ID: 00000002)
Error 4: session copy failed: target session name 'existing_target' is already in use